backend/src/main/java/fk/prof/backend/util/BitOperationUtil.java

```diff
@@ -0,0 +1,9 @@
+package fk.prof.backend.util;
+
+public class BitOperationUtil {
+
+  public static long constructLongFromInts(int mostSignificantBits, int leastSignificantBits) {
+    return (((long)mostSignificantBits) << 32) | (leastSignificantBits & 0xffffffffL);
```

Comment:
This works with negative msb and negative lsb as well. No special checks are required